src/componets/calculator.js

```diff
@@ -1,14 +1,74 @@
-import './calculator.css';
-import PropTypes from 'prop-types';
+import { useState } from 'react';
+import Button from './button';
+import calculate from '../logic/calculate';
 
-const Calculator = ({ Children }) => (
-  <div className="calc-wrapper">
-    { Children }
-  </div>
-);
+export default function Calculator() {
+  const [state, setState] = useState({
+    next: null,
+    operation: null,
+    total: null,
+  });
+  const { next, operation, total } = state;
+  const eventHandler = (e) => {
+    setState(calculate(state, e.target.name));
+  };
 
-Calculator.propTypes = {
-  Children: PropTypes.node.isRequired,
-};
-
-export default Calculator;
+  return (
+    <div className="calculator-container">
+      <div className="calculator">
+        <div className="row">
+          <div className="display">{next || operation || total || 0}</div>
+        </div>
+        <div className="row">
+          <Button className="calc-btn" value="AC" handleClick={eventHandler} />
+          <Button className="calc-btn" value="+/-" handleClick={eventHandler} />
+          <Button className="calc-btn" value="%" handleClick={eventHandler} />
+          <Button
+            className="calc-btn orange"
+            value="÷"
+            handleClick={eventHandler}
+          />
+        </div>
+        <div className="row">
+          <Button className="calc-btn" value="7" handleClick={eventHandler} />
+          <Button className="calc-btn" value="8" handleClick={eventHandler} />
+          <Button className="calc-btn" value="9" handleClick={eventHandler} />
+          <Button
+            className="calc-btn orange"
+            value="*"
+            handleClick={eventHandler}
+          />
```

Comment:
Well done for implementing the events for all buttons. However, I am not able to perform multiplication. Kindly fix it.

![image](https://github.com/techmoves/math-magicians/assets/20554583/95f8f7cd-33ef-404d-9b8a-91d8628a327b)